op-bindings/bindgen/etherscan-contracts.go

```
@@ -0,0 +1,154 @@
+}
+
+type etherscanContractMetadata struct {
+	Name              string
+	DeployedBin       string
+	Package           string
+}
+
+func readEtherscanContractsList(filePath string) ([]etherscanContract, error) {
+	var contracts []etherscanContract
+	err := readJSONFile(filePath, &contracts)
+	return contracts, err
+}
+
+func fetchEtherscanData(url string) (string, error) {
+	// TODO Handle rate limit error
+	resp, err := http.Get(url)
+	if err != nil {
+		return "", err
+	}
+	defer resp.Body.Close()
+
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return "", err
+	}
+
+	var response etherscanApiResponse
+	err = json.Unmarshal(body, &response)
+	if err != nil {
+		return "", err
+	}
+
+	return response.Result, nil
+}
+
+func writeEtherscanContractMetadata(contractMetaData etherscanContractMetadata, metadataOutputDir, contractName string, fileTemplate *template.Template) error {
+	metaDataFilePath := filepath.Join(metadataOutputDir, strings.ToLower(contractName)+"_more.go")
+	metadataFile, err := os.OpenFile(
+		metaDataFilePath,
+		os.O_RDWR|os.O_CREATE|os.O_TRUNC,
+		0o600,
+	)
+	if err != nil {
+		return fmt.Errorf("Error opening %s's metadata file at %s: %v", contractName, metaDataFilePath, err)
+	}
+
+	if err := fileTemplate.Execute(metadataFile, contractMetaData); err != nil {
+		return fmt.Errorf("Error writing %s's contract metadata at %s: %v", contractName, metaDataFilePath, err)
+	}
+	metadataFile.Close()
+
+	log.Printf("Wrote %s's contract metadata to: %s", contractName, metaDataFilePath)
+	return nil
+}
+
+func genEtherscanBindings(contractListFilePath, sourceMapsListStr, etherscanApiKey, goPackageName, metadataOutputDir string) {
+	contracts, err := readEtherscanContractsList(contractListFilePath)
+	if err != nil {
+		log.Fatalf("Error reading contract list %s: %v", contractListFilePath, err)
+	}
+
+	if len(contracts) == 0 {
+		log.Fatalf("No contracts parsable from given contract list: %s", contractListFilePath)
```

We should return errors instead of `log.Fatalf` to make these more portable and then log the errors in the cli